File: admin/src/types/purchase.ts
export interface PurchaseItem {
  product_id: string
  product_name: string
  sku: string
  quantity: number
  unit_cost: number
  subtotal: number
}

export interface Purchase {
  id: string
  supplier_name: string
  items: PurchaseItem[]
  total_amount: number
  paid_amount: number
  status: 'pending' | 'received' | 'cancelled'
  note: string
  created_by: string
  created_at: string
  updated_at: string
}

export interface CreatePurchaseInput {
  supplier_name: string
  items: { product_id: string; quantity: number; unit_cost: number }[]
  paid_amount: number
  note: string
}

File: backend/models/purchase.go
package models

type Purchase struct {
	ID            string         `json:"id"`
	SupplierName  string         `json:"supplier_name"`
	Items         []PurchaseItem `json:"items"`
	TotalAmount   float64        `json:"total_amount"`
	PaidAmount    float64        `json:"paid_amount"`
	PaymentMethod string         `json:"payment_method"` // "cash" | "bkash" | "nagad" | "rocket" | "bank_transfer" | "card"
	TransactionID string         `json:"transaction_id"` // bKash/Nagad/Rocket/Bank TrxID
	Status        string         `json:"status"`         // "pending" | "received" | "cancelled"
	Note          string         `json:"note"`
	CreatedBy     string         `json:"created_by"`
	CreatedAt     string         `json:"created_at"`
	UpdatedAt     string         `json:"updated_at"`
}

type PurchaseItem struct {
	ProductID   string  `json:"product_id"`
	ProductName string  `json:"product_name"`
	SKU         string  `json:"sku"`
	Quantity    int     `json:"quantity"`
	UnitCost    float64 `json:"unit_cost"`
	Subtotal    float64 `json:"subtotal"`
}
